packages/flutter_tools/test/general.shard/compile_expression_test.dart

```diff
@@ -50,15 +48,13 @@ void main() {
       fileSystem: fileSystem,
     );
 
-    when(mockFrontendServer.stdin).thenReturn(mockFrontendServerStdIn);
+    stdErrStreamController = StreamController<String>();
+    when(mockFrontendServer.stdin).thenReturn(frontendServerStdIn);
     when(mockFrontendServer.stderr)
-        .thenAnswer((Invocation invocation) => mockFrontendServerStdErr);
+        .thenAnswer((Invocation invocation) => stdErrStreamController.stream.transform(utf8.encoder));
     when(mockFrontendServer.exitCode).thenAnswer((Invocation invocation) {
       return Completer<int>().future;
     });
-    stdErrStreamController = StreamController<String>();
-    when(mockFrontendServerStdErr.transform<String>(any))
-        .thenAnswer((Invocation invocation) => stdErrStreamController.stream);
```

Comment:
This extra `MockStream` redirection was unnecessary, directly look up `mockFrontendServer.stderr` to the `stdErrStreamController` `StreamController`.